tests/unit/models/QuestionTest.php

```diff
@@ -44,4 +44,94 @@ public function testGetQuestionThemeOnSavedQuestion()
         $this->assertNotEmpty($questionTheme);
         $this->assertEquals("bootstrap_buttons_multi", $questionTheme->name);
     }
+
+    /**
+     * Test validating a question theme name
+     * when question type is set to null.
+     */
+    public function testValidateQuestionThemeNameNoType()
+    {
+        $question = new Question();
+        $question->type = null;
+
+        $result = $question->questionThemeNameValidator();
+        $this->assertNull($result, 'An invalid validation result was expected since the question type was set to null.');
+    }
+
+    /**
+     * Test validating a question theme name
+     * when evaluating a child question.
+     */
+    public function testValidateChildQuestionThemeName()
+    {
+        $question = new Question();
+        $question->parent_qid = 1;
+
+        $result = $question->questionThemeNameValidator();
+        $this->assertNull($result, 'An invalid validation result was expected since a child question is being validated.');
+    }
+
+    /**
+     * Test getting a default theme name from type
+     */
+    public function testGetDefaultQuestionThemeNameFromType()
+    {
+        $question = new Question();
+
+        $themeName = $question->questionThemeNameValidator();
+        $this->assertSame('longfreetext', $themeName, 'Incorrect default question theme name.');
+    }
+
+    /**
+     * Test getting a default theme name from an invalid type.
+     */
+    public function testGetDefaultQuestionThemeNameFromInvalidType()
+    {
+        $question = new Question();
+        $question->type = 'Test';
+
+        $themeName = $question->questionThemeNameValidator();
+        $this->assertNull($themeName, 'An invalid validation result was expected since "Z" question type does not exist.');
+    }
+
+    /**
+     * Test getting the core theme name.
+     */
+    public function testGetCoreQuestionThemeName()
+    {
+        $question = new Question();
+        $question->type = "M";
+        $question->question_theme_name = 'core';
+
+        $themeName = $question->questionThemeNameValidator();
+        $this->assertSame('multiplechoice', $themeName, 'Incorrect core question theme name.');
```

Comment:
It's not core now ? I don't rememeber  …

Reply:
I think so. I mean, it is not failing. Let'wait and see. I don't recall either.

Reply:
There are update on QuestionTheme system since i create the PR …

@ptelu you work on the new QuetsionTheme system ?

Reply:
>  Maybe related to some refactoring that Kevin, Tim and Patricia are doing

Some month ago : core theme are set to `core` in DB. Unsure when it was updated too. But you wrok on it 5 years before … 
https://github.com/LimeSurvey/LimeSurvey/commit/7d977047c6b6773cf2e8e531c3299f79851a9a1d 😭

It's done via https://github.com/LimeSurvey/LimeSurvey/blob/b1146f3f31437c3b4891a74418041dc1441b76d8/application/core/LsDefaultDataSets.php#L1860

Reply:
> shouldnt this be done by finding the core theme based on "core_theme" and "question_type" column in question_themes table?

Why? Don't we already know we expect "multiplechoice" for that combination of `type` and `uestion_theme_name`?

Reply:
Yep :)

Some more DEV issue to report, unsure on desired behaviour

- disable core question theme and allow only user question theme
- disable core question theme without user question theme (i have it for upload question type)